### PR TITLE
update instrument calibrator and runner

### PR DESCRIPTION
**Pull request by joelvbernier**
Fixes: #855 
Fixes: #859

### CODE REVIEW

File: hexrd/ui/calibration/auto/instrument_calibrator.py [hunk 1]
```diff
@@ -64,22 +155,25 @@ def run_calibration(self,
...
+                print('OPTIMIZATION SUCCESSFUL!!!')
             else:
                 print('no improvement in residual!!!')
-                step_successful = False
+            print('initial ssr: '
+                  f'{sum(resd0**2)/float(len(resd0))}')
+            print('final ssr: '
+                  f'{sum(resd1**2)/float(len(resd1))}')
 
+            step_successful = False
```
Comment (psavery) [inline hexrd/ui/calibration/auto/instrument_calibrator.py:176]: Looks like this is causing the loop to end after 1 iteration, is that correct?

File: hexrd/ui/calibration/auto/instrument_calibrator.py [hunk 2]
```diff
@@ -19,43 +41,112 @@ def instr(self):
...
+        master_data_dict_list = []
+        for calib_class in self.calibrators:
+            master_data_dict_list.append(
+                calib_class._extract_powder_lines(
+                    fit_tth_tol=fit_tth_tol, int_cutoff=int_cutoff
+                )
+            )
+        return master_data_dict_list
+
+    def residual(self, x0, master_data_dict_list):
```
Comment (psavery) [inline hexrd/ui/calibration/auto/instrument_calibrator.py:89]: Looks like the argument `x0` is not being used. Is that okay?

Comment (joelvbernier) [reply hexrd/ui/calibration/auto/instrument_calibrator.py:89]: Nope!   Fixed.

File: hexrd/ui/calibration/auto/instrument_calibrator.py (hunk 1 above)
Comment (joelvbernier) [reply hexrd/ui/calibration/auto/instrument_calibrator.py]: bad indent; fixed.

**Review by psavery — APPROVED** [review]
LGTM

**Pull request merged**